mathjax3-ts/output/chtml/Wrappers/mfrac.ts

```diff
@@ -140,27 +235,171 @@ export class CHTMLmfrac<N, T, D> extends CHTMLWrapper<N, T, D> {
     }
 
     /*
-     * @override
+     * @param{BBox} bbox        The buonding box to modify
+     * @param{boolean} display  True for display-mode fractions
+     * @param{number} t         The thickness of the line
      */
-    public computeBBox(bbox: BBox) {
-        bbox.empty();
-        const attr = this.node.attributes.getList('displaystyle', 'scriptlevel');
-        const display = attr.displaystyle && attr.scriptlevel === 0;
+    protected getFractionBBox(bbox: BBox, display: boolean, t: number) {
         const nbox = this.childNodes[0].getBBox();
         const dbox = this.childNodes[1].getBBox();
-        const pad = (this.node.getProperty('withDelims') as boolean ? 0 : this.font.params.nulldelimiterspace);
-        const a = this.font.params.axis_height;
-        const T = (display ? 3.5 : 1.5) * this.font.params.rule_thickness;
-        bbox.combine(nbox, 0, a + T + Math.max(nbox.d * nbox.rscale, display ? .217 : .054));
-        bbox.combine(dbox, 0, a - T - Math.max(dbox.h * dbox.rscale, display ? .726 : .505));
+        const fparam = this.font.params;
+        const pad = (this.node.getProperty('texWithDelims') as boolean ? 0 : fparam.nulldelimiterspace);
+        const a = fparam.axis_height;
+        const T = (display ? 3.5 : 1.5) * t;
+        bbox.combine(nbox, 0, a + T + Math.max(nbox.d * nbox.rscale, (display ? fparam.num1 : fparam.num2) - a - T));
+        bbox.combine(dbox, 0, a - T - Math.max(dbox.h * dbox.rscale, (display ? fparam.denom1 : fparam.denom2) + a - T));
```

Comment:
Some computations are the same as in `makeFraction` (`a` and second args of `Math.max`).
Could they be combined into an aux method, thereby shortening some of these lines?